src/sdl2_compat.c

```diff
@@ -6225,12 +6226,32 @@ SDL_QueryTexture(SDL_Texture *texture, Uint32 *format, int *access, int *w, int
 SDL_DECLSPEC int SDLCALL
 SDL_SetTextureScaleMode(SDL_Texture *texture, SDL_ScaleMode scaleMode)
 {
+    SDL_PropertiesID texture_properties = SDL3_GetTextureProperties(texture);
+    SDL3_SetNumberProperty(texture_properties, PROP_TEXTURE_SCALE_MODE, scaleMode);
```

Comment:
It should probably only do this if setting the scale mode succeeds.

Reply:
I'm thinking to replace the properties with a linked list.
There's currently a leak in SDL_quit when not all hooked callbacks are removed.